### PR TITLE
Pointersimulation

### PR DESCRIPTION
**Pull request by sebllll**
# PR Details

just adds adds PointerSimulated.cs. That class is analogue to MouseSimulated.cs

## Description

This is still a WIP and also my first PR to stride, so please provide feedback on what to improve or change.
One thing i was unsure is the GUID: 
i created the GUI via VS and just inserted it here. Maybe something more has to be done with this GUID like registering it somewhere else.

## Motivation and Context

It helps me to create pointer events from various sources. Imagine a special input-device like a touchframe, that doesn't come with a touch-driver but sends its events via network. With this class, you can create an easy adapter, that translates the incoming network events to pointer-events.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: sources/engine/Xenko.Input/Simulated/InputSourceSimulated.cs [hunk 1]
```diff
@@ -15,11 +15,14 @@ public class InputSourceSimulated : InputSourceBase
...
         private List<KeyboardSimulated> keyboards = new List<KeyboardSimulated>();
-        
+        private List<PointerSimulated> pointerDevices = new List<PointerSimulated>();
+
+
         public IReadOnlyList<KeyboardSimulated> Keyboards => keyboards;
         public IReadOnlyList<MouseSimulated> Mice => mice;
         public IReadOnlyList<GamePadSimulated> GamePads => gamePads;
-        
+        public IReadOnlyList<PointerSimulated> PointerDevices => pointerDevices;
```
Comment (xen2) [inline sources/engine/Xenko.Input/Simulated/InputSourceSimulated.cs]: If we want to follow name convention of other devices, maybe `PointerDevices` should be renamed `Pointers`.

File: sources/engine/Xenko.Input/Simulated/InputSourceSimulated.cs [hunk 2]
```diff
@@ -15,11 +15,14 @@ public class InputSourceSimulated : InputSourceBase
         private List<GamePadSimulated> gamePads = new List<GamePadSimulated>();
         private List<MouseSimulated> mice = new List<MouseSimulated>();
         private List<KeyboardSimulated> keyboards = new List<KeyboardSimulated>();
-        
+        private List<PointerSimulated> pointerDevices = new List<PointerSimulated>();
```
Comment (xen2) [inline sources/engine/Xenko.Input/Simulated/InputSourceSimulated.cs]: If we want to follow name convention of other devices, maybe `pointerDevices` should be renamed `pointers`.

File: sources/engine/Xenko.Input/Simulated/InputSourceSimulated.cs [hunk 3]
```diff
@@ -101,6 +105,29 @@ public void RemoveAllKeyboards()
             keyboards.Clear();
         }
 
+        public PointerSimulated AddPointerDevice()
```
Comment (xen2) [inline sources/engine/Xenko.Input/Simulated/InputSourceSimulated.cs]: Same, `AddPointer`, `RemovePointer`, etc.

File: sources/engine/Xenko.Input/Simulated/PointerSimulated.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Xenko contributors (https://xenko.com) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
```
Comment (xen2) [inline sources/engine/Xenko.Input/Simulated/PointerSimulated.cs]: New file headers should not contain Silicon Studio Corp copyright, only xenko part

**Comment by sebllll** [comment]
ok thanks, I did the changes.

also removed this unnecessary enum from InputSourceSimulated.cs:
```csharp
private enum DeviceEventType
{
    Add,
    Remove,
}
```

**Comment by xen2** [comment]
Merged, thanks!

**Pull request merged**